www/__tests__/timelineHelper.test.ts

```diff
@@ -0,0 +1,60 @@
+import { clearAlerts, mockAlert, mockLogger } from '../__mocks__/globalMocks';
+import { readAllCompositeTrips, readUnprocessedTrips } from '../js/diary/timelineHelper';
+import { mockBEMUserCache } from '../__mocks__/cordovaMocks';
+
+import * as mockTLH from '../__mocks__/timelineHelperMocks';
+
+mockLogger();
+mockAlert();
+mockBEMUserCache();
+
+beforeEach(() => {
+  clearAlerts();
+});
+
+afterAll(() => {
+  jest.restoreAllMocks();
+});
+
+// Once we have end-to-end testing, we could utilize getRawEnteries.
+jest.mock('../js/commHelper', () => ({
+  getRawEntries: jest.fn((key, startTs, endTs, valTwo) => {
+    if (startTs === mockTLH.fakeStartTsOne) return mockTLH.mockData;
+    if (startTs == mockTLH.fakeStartTsTwo) return mockTLH.mockDataTwo;
+    return {};
+  }),
+}));
+
+it('works when there are no composite trip objects fetched', async () => {
+  expect(readAllCompositeTrips(-1, -1)).resolves.toEqual([]);
+});
+
+it('fetches a composite trip object and collapses it', async () => {
+  expect(readAllCompositeTrips(mockTLH.fakeStartTsOne, mockTLH.fakeEndTsOne)).resolves.toEqual(
+    mockTLH.readAllCheckOne,
```

Comment:
For something like this, I don't know that you need to compare the entire result against some 'check' object (this is commonly called a 'snapshot')

I think you can just look through the result and ensure that it has the values you would expect based on what you know the function you're testing is supposed to do.

In this instance, `readAllCompositeTrips` is supposed to input what's received from the server and output an array of unpacked `CompositeTrip`s.
So I would check that the result is an array of objects with a length of however many you expect from the mock. Each of these objects has `start_ts` / `end_ts` / any other distinctive properties that would define them as `CompositeTrip`s. The objects should not have `data` / `metadata` because they've been unpacked by now. They **should** have `_id`, `key`, and `origin_key` because these properties get mixed in during unpacking.
Then I would dig into the trips' `start_confirmed_place` and `end_confirmed_place` and make sure they've also been unpacked.